lib/modules/data_store.rb

```diff
@@ -0,0 +1,28 @@
+module DataStore
+  def add_item
+  end
+
+  def append_entry
+  end
+
+  def get_item(item_hash)
+  end
+
+  def get_items
+  end
+
+  def get_entry(entry_type, entry_number)
```

Comment:
I'm starting to think that these should be in terms of index_name (i.e. "record" or "metadata") rather than entry_type. We essentially then just map `IndexName "record" == EntryType "user"` and `IndexName "metadata" == EntryType "system"` on the DataStore side. 

I will give this some more thought.

Reply:
This makes sense until we combine the current separate entry tables into, at which point we no longer need to make this distinction between whether the entry is of type `user` or `system`, as entry numbers are then once again unique.

Reply:
Yep, agree

Reply:
@michaelabenyohai could you elaborate on these equalities? Also, what is the benefit of doing this vs keeping it to entry?